File: modules/apps/site-initializer/site-initializer-raylife-d2c/extra/remote-app/src/routes/selected-quote/services/Channel.js
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import {axios} from '../../../common/services/liferay/api';

const DeliveryAPI = 'o/headless-commerce-admin-channel';
const ChannelName = 'Raylife D2C Channel';

export function getChannel() {
	return axios.get(
		`${DeliveryAPI}/v1.0/channels?filter=name eq '${ChannelName}'`
	);
}

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/model/Channel.java
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.commerce.product.definitions.web.internal.model;

/**
 * @author Alessio Antonio Rendina
 */
public class Channel {

	public Channel(long commerceChannelRelId, String name) {
		_commerceChannelRelId = commerceChannelRelId;
		_name = name;
	}

	public long getCommerceChannelRelId() {
		return _commerceChannelRelId;
	}

	public String getName() {
		return _name;
	}

	private final long _commerceChannelRelId;
	private final String _name;

}